arviz/plots/backends/matplotlib/distplot.py

```diff
@@ -87,12 +88,22 @@ def plot_dist(
 
 def _histplot_mpl_op(values, values2, rotated, ax, hist_kwargs):
     """Add a histogram for the data to the axes."""
+    if isinstance(values, xr.Dataset):
+        raise ValueError(
+            "Cannot directly convert xarray.Dataset to numpy array."
+            " Instead, create an xarray.DataArray first "
+            "or Use plot_posterior, plot_density, plot_joint"
+            "or plot_pair instead of plot_dist"
+        )
+    if isinstance(values, InferenceData):
+        raise ValueError(" Inference Data object detected. Use plot_posterior instead of plot_dist")
+
```

Comment:
This should go in `plots/distplot.py`

Reply:
Yes, we will have to add it to both. Will do.

Reply:
Not to both, only to `arviz/plots/distplot.py`